python/pyspark/errors/exceptions/captured.py

```diff
@@ -185,6 +186,11 @@ def getQueryContext(self) -> List[BaseQueryContext]:
 
 
 def convert_exception(e: "Py4JJavaError") -> CapturedException:
+    converted = _convert_exception(e)
+    return recover_python_exception(converted)
```

Comment:
If this is specific to UDF, I think we can do this below for `org.apache.spark.api.python.PythonException` case. For connect case too.